src/hotspot/cpu/aarch64/templateInterpreterGenerator_aarch64.cpp

```diff
@@ -832,6 +832,7 @@ void TemplateInterpreterGenerator::generate_fixed_frame(bool native_call) {
   __ ldr(rcpool, Address(rcpool, ConstantPool::cache_offset_in_bytes()));
   __ stp(rlocals, rcpool, Address(sp, 2 * wordSize));
 
+  __ protect_return_address();
```

Comment:
Most of the changes to fix the test look fairly self-explanatory but I don't really understand why you relocated call to protect_return-_address from its previous location at line 801. Can you explain why it has been moved?

Reply:
I originally moved it as part of debugging (a GC load_at occurs during the load_mirror).
Once all the GC changes went in (all the enter_subframe calls), this change was no longer required.
Then, when I came to change it back, I realised it made more sense in the new place. The protect is now directly before the storing of lr to the stack. That's logically a better place and should make the assembler easier to read.

Reply:
Ok, I thought the new place looked better but was not clear why it had not been there in the first place. Thanks for clarifying.